roxie/roxiemem/roxiemem.cpp

```diff
@@ -354,6 +421,7 @@ static void initializeHeap(bool allowHugePages, bool allowTransparentHugePages,
                     //If we notify heapBlockSize items at a time it will always be a multiple of hugePageSize so shouldn't trigger defragmentation
                     heapNotifyUnusedEachBlock = !retainMemory;
                 }
+                heapHugePagesPossible = true;
```

Comment:
Very obscure (would occur on power8 with current configuration)

    else
        origHeapRetainMemory = true;

otherwise after unlocking it would start returning pages.